### PR TITLE
Issue #2174 Handle unregistration properly for rhel iso

### PR DESCRIPTION
**Pull request by anjannath**
Fixes: #2174 

Ref: https://issues.jboss.org/projects/CDK/issues/CDK-270

### CODE REVIEW

File: pkg/minishift/registration/redhat.go [hunk 1]
```diff
@@ -130,6 +130,14 @@ func (registrator *RedHatRegistrator) Unregister(param *RegistrationParameters)
 			"sudo -E subscription-manager unregister"); err != nil {
 			return err
 		}
+		if _, err := registrator.SSHCommand(
+			"sudo -E subscription-manager remove --all"); err != nil {
```
Comment (praveenkumar) [inline pkg/minishift/registration/redhat.go]: Why we need this as per subscription manager doc, it clearly indicates that `unregister` should strip its subscriptions. 

- https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html/quick_registration_for_rhel/un-registering

Comment (LalatenduMohanty) [reply pkg/minishift/registration/redhat.go]: Copying the error from @anjannath 
```
$ sudo -E subscription-manager status
Consumer profile "67845724-d9be-4048-95fa-e81a162c8761" has been deleted from the server. You can use command clean or unregister to remove local profile.
```
So my guess is it is related  to the `local profile` not being cleaned properly

**Comment by LalatenduMohanty** [comment]
@anjannath Any update from Ondrej?

**Comment by anjannath** [comment]
@gbraad Clean removes the local profiles for RHSM, this was added to care for the case where, the registration is canceled in the RHSM server but local profiles are present, it is going to use the old info for again for registering, which i believe caused the issue.

**Comment by anjannath** [comment]
@LalatenduMohanty Onderj have replied on the jira issue, seems like he's seeing some regression, so i rebased the changes again on top of latest release and given him the binary and iso to test..

**Comment by anjannath** [comment]
Here's the latest [comment](https://issues.jboss.org/browse/CDK-270?focusedCommentId=13553753&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13553753) on the jira issue. The binary which he's referring to, which i gave him last, had the registration fix in it. We are good to merge this i guess..

**Comment by budhrg** [comment]
@anjannath He mentioned `Although, registration issue was still there.` in the comment. Not sure it means it has been fixed.

**Comment by anjannath** [comment]
@budhrg He said that for the beta bits, which do not have the fix so registration issue was there... what he confirmed for in the beta bits is that the regression which he was seeing before is not there anymore..

**Comment by budhrg** [comment]
@anjannath 
> that's what i understood from his comment, we could ask him to further clarify this..

Sure. Clarification is needed.

**Comment by LalatenduMohanty** [comment]
@praveenkumar It is waiting for your review. Please review.

File: pkg/minishift/registration/redhat.go [hunk 2]
```diff
@@ -130,6 +130,10 @@ func (registrator *RedHatRegistrator) Unregister(param *RegistrationParameters)
 			"sudo -E subscription-manager unregister"); err != nil {
 			return err
 		}
+		if _, err := registrator.SSHCommand(
+			"sudo -E subscription-manager clean"); err != nil {
+			return err
+		}
```
Comment (praveenkumar) [inline pkg/minishift/registration/redhat.go]: :+1:

**Pull request merged**